### PR TITLE
Feature/dwr 1060 migrate olap

### PR DESCRIPTION
**Pull request by agorina**
This completes all the Steps ITs, but there are still a few outstanding items in this JIRA to be done.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -25,7 +25,7 @@ apply plugin: 'io.spring.dependency-management'
 // Set version dependencies. This allows the cli to force version, e.g. `-Pcommon=1.0.0-SNAPSHOT`
 
 String rdwCommonVersion = project.hasProperty("common") ? project.property("common") : "1.1.0-42"
-String rdwSchemaVersion = project.hasProperty("schema") ? project.property("schema") : "1.1.0-224"
```
Comment (agorina) [inline build.gradle]: I had to make a change to the schema to allow for NULL gender_id.

File: migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/MigrateOlapReportingConfiguration.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public Job migrateReportingJob(final Step cleanupLandingLocationStep,
                                    final Step upsertFactStudentIcaExamStep,
                                    final Step deleteEntitiesStep,
                                    final Step deleteCodesStep,
-                                   final Step upsertStudentsAndGroupsStep,
```
Comment (agorina) [inline migrate-olap/src/main/java/org/opentestsystem/rdw/ingest/migrate/olap/MigrateOlapReportingConfiguration.java]: We are for not migrating `students` only.

File: migrate-olap/src/main/resources/application.yml [hunk 3]
```diff
@@ -42,7 +42,7 @@ migrate:
       role: ${rdw.migrate.aws.redshift.role}
 
   batch:
-    size: 100000
+    size: 500000
```
Comment (agorina) [inline migrate-olap/src/main/resources/application.yml]: This changed the run time (from aws dev) from 3.5 hours to 48 min.

File: migrate-olap/src/test/java/org/opentestsystem/rdw/ingest/migrate/olap/step/UpsertIcaFactExamsIT.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.opentestsystem.rdw.ingest.migrate.olap.step.OlapReportingMigrateStepsConfig.upsertFactStudentIcaExamStepName;
+import static org.opentestsystem.rdw.migrate.common.TableTestCountHelper.verifyTableCountsAfterTest;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.BEFORE_TEST_METHOD;
+
+@Ignore
```
Comment (agorina) [inline migrate-olap/src/test/java/org/opentestsystem/rdw/ingest/migrate/olap/step/UpsertIcaFactExamsIT.java]: Still ignored.

File: migrate-olap/src/test/java/org/opentestsystem/rdw/ingest/migrate/olap/step/UpsertStudentsStepIT.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+import static com.google.common.collect.Lists.newArrayList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.opentestsystem.rdw.ingest.migrate.olap.step.OlapReportingMigrateStepsConfig.upsertStudentsStepName;
+import static org.opentestsystem.rdw.migrate.common.TableTestCountHelper.verifyTableCountsAfterTest;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.AFTER_TEST_METHOD;
+import static org.springframework.test.context.jdbc.Sql.ExecutionPhase.BEFORE_TEST_METHOD;
+
+@Ignore
```
Comment (agorina) [inline migrate-olap/src/test/java/org/opentestsystem/rdw/ingest/migrate/olap/step/UpsertStudentsStepIT.java]: Ditto.

File: migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/UpsertStudentsAndGroupsIT.java [hunk 6]
```diff
@@ -69,7 +69,7 @@ public void itShouldUpdateAndInsert() {
             @Sql(executionPhase = AFTER_TEST_METHOD, scripts = {"classpath:ReportingAndStagingCodesTeardown.sql"}, config = @SqlConfig(dataSource = "reportingDatasource"))
     })
     @Test
-    public void itShouldInsertButNoteDelete() {
+    public void itShouldInsertButNotDelete() {
```
Comment (agorina) [inline migrate-reporting/src/test/java/org/opentestsystem/rdw/ingest/migrate/reporting/step/UpsertStudentsAndGroupsIT.java]: Just noticed a typo.

File: migrate-olap/src/main/resources/application.yml (hunk 3 above)
Comment (malaffoon) [reply migrate-olap/src/main/resources/application.yml]: Clearly that is near-linear ... you should set it to 10,000,000 and it will take 2 minutes?

Comment (agorina) [reply migrate-olap/src/main/resources/application.yml]: Yes, this is pretty much what it took when I did the entire dump. 
But I am afraid of the jdbc timing out.

**Pull request merged**